### DISCUSSION TITLE
Contract-to-contract read of App Global State

### DISCUSSION THREAD

**Discussion by xarmian** [reach-sh/reach-lang#1579] [General]
Is it possible/feasible to add a mechanism to read the app global state of a remote contract from a Reach contract? In this specific case I'm trying to work out a proof of concept to read data from price-pair contracts produced by Goracle, such as this one, where they expose the price and decimals of a token pair as global key/value pairs:

https://testnet.algoexplorer.io/application/70117029

If I'm understanding their business model correctly, they'll have some contracts that will use this approach of exposing data in global state, and others that will utilize an ABI.

**Comment by jeapostrophe** [comment]
It is possible to do this, I could make `ALGO.unsafeReadGlobalState(Contract, Key)`

**Reply by xarmian** [reply]
Yes, something like this would be perfect
